src/govuk/components/task-list/_index.scss

```diff
@@ -0,0 +1,120 @@
+  $govuk-task-list-focus-colour: govuk-colour("black");
+  $govuk-task-list-focus-background-colour: govuk-colour("yellow");
+  $govuk-task-list-hover-colour: govuk-colour("light-grey", $legacy: "grey-3");
+
+  .govuk-task-list {
+    @include govuk-font($size: 19);
+    margin-top: 0;
+    margin-bottom: 0;
+    padding: 0;
+    list-style-type: none;
+    @include govuk-media-query($from: tablet) {
+      min-width: 550px;
+    }
+  }
+
+  // This uses table layout so that the task name and status always appear side-by-side, with the width of
+  // each 'column' being flexible depending upon the length of the task names and statuses.
+  //
+  // The position is set to 'relative' so than an absolutely-positioned transparent element box
+  // can be added within the link so that the whole row can be clickable.
+  .govuk-task-list__item {
+    display: table;
+    position: relative;
+    width: 100%;
+    margin-bottom: 0;
+    padding-top: govuk-spacing(2);
+    padding-bottom: govuk-spacing(2);
+    border-bottom: 1px solid $govuk-border-colour;
+  }
+
+  .govuk-task-list__item:first-child {
+    border-top: 1px solid $govuk-border-colour;
+  }
+
+  // This class is added to the <li> elements where the task name is a link.
+  // The background hover colour is added to help indicate that the whole row is clickable, rather
+  // than just the visible link text.
+  .govuk-task-list__item--with-link:hover {
+    background: $govuk-task-list-hover-colour;
+  }
+
+  .govuk-task-list__task-name-and-hint {
+    display: table-cell;
+    vertical-align: top;
+  }
+
+  .govuk-task-list__status-container {
+    display: table-cell;
+    padding-left: govuk-spacing(2);
+    text-align: right;
+    vertical-align: top;
+  }
+
+  // This adds an empty transparent box covering the whole row, including the task status and
+  // any hint text. Because this is generated within the link element, this allows the whole area
+  // to be clickable.
+  .govuk-task-list__link:after {
+    content: "";
+    display: block;
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+  }
+
+  .govuk-task-list__task_hint {
+    margin-top: govuk-spacing(1);
+    color: $govuk-secondary-text-colour;
+  }
+
+  .govuk-task-list__status {
+    display: inline-block;
+    margin-top: 0;
+    margin-bottom: 0;
+    padding-top: 2px;
+    padding-right: 4px;
+    padding-bottom: 2px;
+    padding-left: 4px;
+  }
+
+  .govuk-task-list__status--light-blue {
+    color: govuk-shade(govuk-colour("blue"), 60);
+    background-color: govuk-tint(govuk-colour("blue"), 90);
+  }
+
+  .govuk-task-list__status--blue {
+    color: govuk-shade(govuk-colour("blue"), 60);
+    background-color: govuk-tint(govuk-colour("blue"), 75);
+  }
+
+  .govuk-task-list__status--text-colour {
+    color: $govuk-text-colour;
+  }
+
+  .govuk-task-list__status--secondary-text-colour {
+    color: $govuk-secondary-text-colour;
+  }
+
+  .govuk-task-list__status--purple {
```

Comment:
I would expect to see class names representing statuses (eg `--incomplete` and `--completed`) rather than colours.

Reply:
@sussexrick Interesting! We deliberated a bit on this, and in the end decided to use colour based names to allow for maximum flexibility for services to use them as they see fit (without it clashing with the class name) but to give some recommendations of how to pair colours with labels in the guidance.

The exception was the red colour, which we decided to give the class name of `govuk-task-list__status--error` to more strongly discourage people from using it for anything other than an error.

All still open to discussion though!